internal/codespaces/api/api.go

```diff
@@ -60,31 +60,40 @@ const (
 
 // API is the interface to the codespace service.
 type API struct {
-	client       httpClient
-	vscsAPI      string
+	client       func() (*http.Client, error)
 	githubAPI    string
 	githubServer string
 	retryBackoff time.Duration
 }
 
-type httpClient interface {
-	Do(req *http.Request) (*http.Response, error)
-}
-
 // New creates a new API client connecting to the configured endpoints with the HTTP client.
-func New(serverURL, apiURL, vscsURL string, httpClient httpClient) *API {
-	if serverURL == "" {
-		serverURL = githubServer
-	}
+func New(f *cmdutil.Factory) *API {
+	apiURL := os.Getenv("GITHUB_API_URL")
```

Comment:
Could we make this and the server URL consts like the URLs above?

Reply:
Looking at other usages of `os.Getenv` in this repo, almost all of them use an inlined string rather than a separate const. I'd rather follow the conventions of the rest of the codebase. I also personally find it easier to read the code with fewer layers of indirection.

https://github.com/search?q=repo%3Acli%2Fcli%20os.Getenv&type=code